packages/roosterjs-content-model-plugins/lib/tableEdit/editors/features/TableMover.ts

```diff
@@ -129,3 +157,212 @@ function isTableTopVisible(editor: IEditor, rect: Rect | null, contentDiv?: Node
+        if (element && domHelper.isNodeInEditor(element)) {
+            return { node: element, offset: 0 };
+        }
+    }
+
+    return null;
+}
+
+function onDragging(context: TableMoverContext, event: MouseEvent, initValue: TableMoverInitValue) {
+    const { tableRect } = initValue;
+    const { editor } = context;
+
+    // Move table outline rectangle
+    tableRect.style.top = `${event.clientY + TABLE_MOVER_LENGTH}px`;
+    tableRect.style.left = `${event.clientX + TABLE_MOVER_LENGTH}px`;
+
+    event.ctrlKey || event.metaKey
+        ? setTableMoverCursor(editor, true, 'copy')
+        : setTableMoverCursor(editor, true, 'move');
+
+    const pos = getNodePositionFromEvent(editor, event.clientX, event.clientY);
+    if (pos) {
+        const range = editor.getDocument().createRange();
+        range.setStart(pos.node, pos.offset);
+        range.collapse(true);
+
+        editor.setDOMSelection({ type: 'range', range, isReverted: false });
+        return true;
+    }
+    return false;
+}
+
+function onDragEnd(
+    context: TableMoverContext,
+    event: MouseEvent,
+    initValue: TableMoverInitValue | undefined
+) {
+    const { editor, table, onFinishDragging: selectWholeTable } = context;
+    const element = event.target;
+    // Remove table outline rectangle
+    initValue?.tableRect.remove();
+
+    // Reset cursor
+    setTableMoverCursor(editor, false);
+
+    if (element == context.div) {
+        // Table mover was only clicked, select whole table
+        selectWholeTable(table);
+        context.onEnd();
+        return true;
+    } else {
+        // Check if table was dragged on itself or Check if element is not in editor
+        if (
+            table.contains(element as Node) ||
+            !editor.getDOMHelper().isNodeInEditor(element as Node)
+        ) {
+            editor.setDOMSelection(initValue?.initialSelection ?? null);
+            context.onEnd();
+            return false;
+        }
+
+        let insertionSuccess: boolean = false;
+
+        // Get position to insert table
+        const insertPosition = getNodePositionFromEvent(editor, event.clientX, event.clientY);
+        if (insertPosition) {
+            // Move table to new position
+            formatInsertPointWithContentModel(
+                editor,
+                insertPosition,
+                (model, context, ip) => {
+                    // Remove old table if not copying
+                    if (!(event.ctrlKey || event.metaKey)) {
+                        const [oldTable, path] = getFirstSelectedTable(model);
+                        if (oldTable) {
+                            const index = path[0].blocks.indexOf(oldTable);
+                            path[0].blocks.splice(index, 1);
+                        }
+                    }
+                    if (ip && initValue?.cmTable) {
+                        // Guarantee the cm Insertion point is selected
+                        ip.marker.isSelected = true;
```

Comment:
Since you are passing in insertPosition into mergeModel at line 323 (which is good), no need to set it to be selected here.